src/java/com/rapleaf/jack/queries/GenericQueryBuilder.java

```diff
@@ -0,0 +1,176 @@
+package com.rapleaf.jack.queries;
+
+import java.io.IOException;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLRecoverableException;
+import java.util.Set;
+
+import com.rapleaf.jack.BaseDatabaseConnection;
+import com.rapleaf.jack.Column;
+import com.rapleaf.jack.Table;
+import com.rapleaf.jack.queries.where_operators.IWhereOperator;
+
+public class GenericQueryBuilder {
+  private static int MAX_CONNECTION_RETRIES = 1;
+
+  private final GenericQuery genericQuery;
+  private final BaseDatabaseConnection dbConnection;
+
+  public GenericQueryBuilder(BaseDatabaseConnection dbConnection, GenericQuery genericQuery) {
```

Comment:
This is one of the places where I struggled the most. You are right that it will be a cleaner design to have the `GenericQueryBuilder` simply build the query. And the `GenericQuery` executes the query.

So maybe something like this:
`GenericQueryBuilder qb = GenericQueryBuilder.from().where(...);`
`Records records = GenericQuery.execute(qb);`

The public static `from()` will be the only method that returns a `GenericQueryBuilder` which ensures only one `FROM` clause for each query.